auth/basic/db.go

```diff
@@ -115,3 +141,39 @@ func (b *Store) Validate(ctx context.Context, username, password string) (string
 
 	return userID, nil
 }
+
+// ValidatePassword should check if the password matches the user's stored password
+func (b *Store) ValidatePassword(ctx context.Context, userID string, password string) (ValidatedPassword, error) {
+	err := validate.Many(
```

Comment:
Do permissions check first before validating. Use `permission.MatchUser` -- this technically validates the password without requiring a username, so instead we're requiring the current request is already authorized for the intended userID.